### PR TITLE
[DDW-1186] Stake pool descriptions is not fully shown if it is long enough

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR fixes stake pool descriptions in not fully shown if it's long enough.

Ticket: https://iohk.myjetbrains.com/youtrack/issue/DDW-1186

## Screenshots

Screenshot 2020-01-30 20 01 54

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1580413579079500)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by a-rukin** [comment]
image
I assume that it would not fit the minumum screen size 🤔 
@nikolaglumac shouldn't we show only first few lines here?

**Comment by IuliaDolishniak** [comment]
this way it looks for Delegation wizard in minimized mode
I know it is not directly by the PR but I think it should be fixed

**Comment by DeeJayElly** [comment]
> image
> 
> I assume that it would not fit the minumum screen size 🤔 @nikolaglumac shouldn't we show only first few lines here?

I can add ellipsis and limit the height of the description text.

**Comment by DeeJayElly** [comment]
> 
> this way it looks for Delegation wizard in minimized mode
> I know it is not directly by the PR but I think it should be fixed

Will recheck @IuliaDolishniak , thats a good catch and definitely it should be fixed

**Comment by a-rukin** [comment]
> > image
> > I assume that it would not fit the minumum screen size 🤔 @nikolaglumac shouldn't we show only first few lines here?
> 
> I can add ellipsis and limit the height of the description text.

https://zpl.io/a8p7ePJ - here is my proposal

**Comment by daniloprates** [comment]
> 
> https://zpl.io/a8p7ePJ - here is my proposal

@a-rukin 
A tooltip over a tooltip looks confusing, to me. I'd try either expanding the text over the other information when hovering or having "Show more"/"Show less" links.

**Comment by a-rukin** [comment]
> > https://zpl.io/a8p7ePJ - here is my proposal
> 
> @a-rukin
> A tooltip over a tooltip looks confusing, to me. I'd try either expanding the text over the other information when hovering or having "Show more"/"Show less" links.

it's tooltip over popup dialog, so quite ok because first requires to click, second - to hover.
In the delegation dialog it's a bit awkward - yes.

As an option instead of "..." we can make gradient fadeout

**Comment by DeeJayElly** [comment]
> > > https://zpl.io/a8p7ePJ - here is my proposal
> > 
> > 
> > @a-rukin
> > A tooltip over a tooltip looks confusing, to me. I'd try either expanding the text over the other information when hovering or having "Show more"/"Show less" links.
> 
> it's tooltip over popup dialog, so quite ok because first requires to click, second - to hover.
> In the delegation dialog it's a bit awkward - yes.
> 
> As an option instead of "..." we can make gradient fadeout

Yeah i understand what you are trying to say. But our component currently we use "pop" as you say, as a tooltip component and for us that's already a popup. That's why @daniloprates said it's strange to see tooltip on top of tooltip.

**Comment by a-rukin** [comment]
Popup over popup is still ok. 
Ex. dropdown menu with multi-levels (if implemented properly)

**Comment by a-rukin** [comment]
Other way around is to create stakepool page. Not sure you are ready to implement it now :D

**Comment by nikolaglumac** [comment]
@DeeJayElly @a-rukin @darko-mijic let's have a call regarding this one today!

**Comment by daniloprates** [comment]
> @DeeJayElly @a-rukin @darko-mijic let's have a call regarding this one today!

@nikolaglumac I'd like to participate as well if it didn't happen yet.

**Comment by DeeJayElly** [comment]
> > @DeeJayElly @a-rukin @darko-mijic let's have a call regarding this one today!
> 
> @nikolaglumac I'd like to participate as well if it didn't happen yet.

Let me know what timeframe? We can talk now or you want to go through it on our weekly call?

**Comment by daniloprates** [comment]
My suggestion: go for @a-rukin's proposal and think of a better overall solution. I have an idea that could integrate this issue with the Stake Pools comparison screen.

**Comment by DeeJayElly** [comment]
@IuliaDolishniak @a-rukin @DmitriiGaico feel free to recheck it now

**Comment by IuliaDolishniak** [comment]
screen
still facing the issue

**Comment by IuliaDolishniak** [comment]
screen2
that's how long Decsription looks like

**Comment by DeeJayElly** [comment]
> screen2
> that's how long Decsription looks like

How did you tested this? If you changed the description text on the fly, directly in the DOM/HTML then that's not the real case scenario because we can't simulate the same behaviour like that.

We need to change the description of the staking tooltip before that element is rendered in order to see the scrollbar.

**Comment by IuliaDolishniak** [comment]
> screen
> still facing the issue

Issue existed in previous versions. Was it agreed in this PR or open another one @nikolaglumac ?

**Comment by IuliaDolishniak** [comment]
> > screen2
> > that's how long Decsription looks like
> 
> How did you tested this? If you changed the description text on the fly, directly in the DOM/HTML then that's not the real case scenario because we can't simulate the same behaviour like that.
> 
> We need to change the description of the staking tooltip before that element is rendered in order to see the scrollbar.

I think we are unable to make these changes, can you make them visible for us while testing?

**Comment by DmitriiGaico** [comment]
> > > screen2
> > > that's how long Decsription looks like
> > 
> > 
> > How did you tested this? If you changed the description text on the fly, directly in the DOM/HTML then that's not the real case scenario because we can't simulate the same behaviour like that.
> > We need to change the description of the staking tooltip before that element is rendered in order to see the scrollbar.
> 
> I think we are unable to make these changes, can you make them visible for us while testing?

To the all the Pools we just need to run Corresponding Rewards Build.

**Comment by DeeJayElly** [comment]
@DmitriiGaico @IuliaDolishniak i added some changes, so now you will be able to see it just by changing the DOM/HTML value in the inspect tool.

**Comment by IuliaDolishniak** [comment]
> @DmitriiGaico @IuliaDolishniak i added some changes, so now you will be able to see it just by changing the DOM/HTML value in the inspect tool.

@DeeJayElly this part is working fine. Previous issue we are leaving as it is?

**Comment by nikolaglumac** [comment]
@IuliaDolishniak there is just that one more issue with tooltip being cut off on small screen. We will discuss it tomorrow.

Are there any other issues?

**Comment by IuliaDolishniak** [comment]
@nikolaglumac no, I am talking regarding this one (cutoff tooltip), I didnt know the reslution on it.

**Comment by nikolaglumac** [comment]
@IuliaDolishniak yes I am talking about the same one. Is that the only issue we have?

**Comment by IuliaDolishniak** [comment]
@nikolaglumac from my side - yes.

**Comment by nikolaglumac** [comment]
Ok. We will discuss what to do with that one tomorrow.

@DmitriiGaico are there any other issues you are aware of?

**Comment by DmitriiGaico** [comment]
> Ok. We will discuss what to do with that one tomorrow.
> 
> @DmitriiGaico are there any other issues you are aware of?

In general the fix looks good. But no approval yet, please don't ask how i found this.

CLSD Pool Production.
CLSD Prod

Same Pool for the PR's Build. Looks like something is wrong. 
CLSD PR

**Comment by nikolaglumac** [comment]
@DeeJayElly please update this line of the code: https://github.com/input-output-hk/daedalus/blob/v2-integration/source/main/config.js#L75 and increase the `MIN_WINDOW_CONTENT_HEIGHT ` to `700`. That will fix the tooltip clipping issues.

Please also remember to update these min-heights in storybook: https://github.com/input-output-hk/daedalus/blob/v2-integration/storybook/stories/_support/config.js#L44-L48 🙏 

@IuliaDolishniak this will resolve the last remaining issue you have reported ☝️

**Comment by nikolaglumac** [comment]
@DeeJayElly please don't forget to check this one too: https://github.com/input-output-hk/daedalus/pull/1832#issuecomment-581623864 
Thanks!
cc @DmitriiGaico

**Comment by DeeJayElly** [comment]
> > Ok. We will discuss what to do with that one tomorrow.
> > @DmitriiGaico are there any other issues you are aware of?
> 
> In general the PR looks good. But no approval yet, please don't ask how i found this.
> 
> CLSD Pool Production.
> CLSD Prod
> 
> Same Pool for the PR's Build. Looks like something is wrong.
> CLSD PR

@DmitriiGaico can you somehow show me how did you managed to reproduce this issue?

**Comment by DmitriiGaico** [comment]
@DeeJayElly Sure, i just installed latest Rewards Build. For that moment it was 10535 and opened CLSD Pool. So no Data Injection or any other Manipulations.

File: storybook/stories/_support/config.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const osNames: Array<any> = Object.keys(operatingSystems);
 export const osMinWindowHeights = {
   Windows: '541px',
```
Comment (nikolaglumac) [inline storybook/stories/_support/config.js]: @DeeJayElly you forgot to update the min-heights for Windows an Linux...

File: source/renderer/app/components/staking/stake-pools/StakePoolTooltip.scss [hunk 2]
```diff
@@ -196,9 +196,14 @@
...
   margin-bottom: 4px;
-  max-height: 79px;
-  overflow: hidden;
-  text-overflow: ellipsis;
+  margin-right: -5px;
+  max-height: 57px;
+  overflow-y: auto;
+  word-break: break-word;
+
+  &.scrollableDescription {
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/stake-pools/StakePoolTooltip.scss]: @DeeJayElly I don't see this CSS class being used anywhere in the code?

File: source/renderer/app/components/staking/stake-pools/StakePoolTooltip.scss [hunk 3]
```diff
@@ -196,9 +196,14 @@
   font-size: 14px;
   line-height: 1.36;
   margin-bottom: 4px;
-  max-height: 79px;
-  overflow: hidden;
-  text-overflow: ellipsis;
+  margin-right: -5px;
+  max-height: 57px;
+  overflow-y: auto;
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/stake-pools/StakePoolTooltip.scss]: @DeeJayElly can you perhaps try the `overflow-y: overlay` option here instead?

Comment (nikolaglumac) [reply source/renderer/app/components/staking/stake-pools/StakePoolTooltip.scss]: @DeeJayElly actually disregard this one - it won't work!

File: storybook/stories/_support/config.js (hunk 1 above)
Comment (DeeJayElly) [reply storybook/stories/_support/config.js]: Done

**Pull request merged**